Assets/MixedRealityToolkit-SDK/Features/UX/Scripts/Tooltips/ToolTip.cs

```diff
@@ -0,0 +1,540 @@
+        public DisplayModeType GroupTipState
+        {
+            set
+            {
+                groupTipState = value;
+            }
+            get
+            {
+                return groupTipState;
+            }
+        }
+
+        [SerializeField]
+        private DisplayModeType masterTipState;
+        /// <summary>
+        /// getter/setter for display state of master tooltip
+        /// </summary>
+        public DisplayModeType MasterTipState
+        {
+            set
+            {
+                masterTipState = value;
+            }
+            get
+            {
+                return masterTipState;
+            }
+        }
+
+        [Tooltip("GameObject that the line and text are attached to")]
+        [SerializeField]
+        private GameObject anchor;
+        /// <summary>
+        /// getter/setter for ameObject that the line and text are attached to
+        /// </summary>
+        public GameObject Anchor
+        {
+            get
+            {
+                return anchor;
+            }
+            set
+            {
+                anchor = value;
+            }
+        }
+
+        [Tooltip("Pivot point that text will rotate around as well as the point where the Line will be rendered to.")]
+        [SerializeField]
+        private GameObject pivot;
+        /// <summary>
+        /// Pivot point that text will rotate around as well as the point where the Line will be rendered to. 
+        /// </summary>
+        public GameObject Pivot
+        {
+            get
+            {
+                return pivot;
+            }
+        }
+
+        [Tooltip("GameObject text that is displayed on the tooltip.")]
+        [SerializeField]
+        private GameObject label;
+
+        [Tooltip("Parent of the Text and Background")]
+        [SerializeField]
+        private GameObject contentParent;
+
+        [Tooltip("Text for the ToolTip to say")]
+        [SerializeField]
+        [TextArea]
+        private string toolTipText;
+        /// <summary>
+        /// Text for the ToolTip to display
+        /// </summary>
+        public string ToolTipText
+        {
+            set
+            {
+                if (value != toolTipText)
+                {
+                    toolTipText = value;
+                    RefreshLocalContent();
+                    if (ContentChange != null)
+                        ContentChange.Invoke();
+                }
+            }
+            get
+            {
+                return toolTipText;
+            }
+        }
+
+        [Tooltip("The padding around the content (height / width)")]
+        [SerializeField]
+        private Vector2 backgroundPadding;
+
+        [Tooltip("The offset of the background (x / y / z)")]
+        [SerializeField]
+        private Vector3 backgroundOffset;
+        /// <summary>
+        /// The offset of the background (x / y / z)
+        /// </summary>
+        public Vector3 LocalContentOffset
```

Comment:
It's a little confusing to have LocalContentOffset backed by backgroundOffset. can we unify the names?